### PR TITLE
Convert ui-admin package to js

### PR DESCRIPTION
**Pull request by MartinSchoeler**
@RocketChat/core

### CODE REVIEW

File: packages/rocketchat-ui-admin/client/admin.js [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
+
+Template.admin.onDestroyed(function() {
+	return TempSettings.remove({});
+});
+
+Template.admin.helpers({
+	languages() {
+		const languages = TAPi18n.getLanguages();
+		let result = [];
+		Object.keys(languages).forEach((key) => {
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: I think one ```.map()``` would be better

File: packages/rocketchat-ui-admin/client/admin.js [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+	group() {
+		const groupId = FlowRouter.getParam('group');
+		const group = RocketChat.settings.collectionPrivate.findOne({
+			_id: groupId,
+			type: 'group'
+		});
+		if (!group) {
+			return;
+		}
+		const settings = RocketChat.settings.collectionPrivate.find({
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: use one line

File: packages/rocketchat-ui-admin/client/admin.js [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+			group: groupId
+		}, {
+			sort: {
+				section: 1,
+				sorter: 1,
+				i18nLabel: 1
+			}
+		}).fetch();
+		const sections = {};
+		Object.keys(settings).forEach((key) => {
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: use ```key => ```

File: packages/rocketchat-ui-admin/client/admin.js [hunk 4]
```diff
@@ -0,0 +1,668 @@
...
+			if (setting.i18nDefaultQuery != null) {
+				if (_.isString(setting.i18nDefaultQuery)) {
+					i18nDefaultQuery = JSON.parse(setting.i18nDefaultQuery);
+				} else {
+					i18nDefaultQuery = setting.i18nDefaultQuery;
+				}
+				if (!_.isArray(i18nDefaultQuery)) {
+					i18nDefaultQuery = [i18nDefaultQuery];
+				}
+				Object.keys(i18nDefaultQuery).forEach((key) => {
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: use ```key => ```

File: packages/rocketchat-ui-admin/client/admin.js [hunk 5]
```diff
@@ -0,0 +1,668 @@
...
+			}
+			const settingSection = setting.section || '';
+			if (sections[settingSection] == null) {
+				sections[settingSection] = [];
+			}
+			sections[settingSection].push(setting);
+		});
+
+		group.sections = [];
+		Object.keys(sections).forEach((key) =>{
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: use ```key => ```

Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: use a map instead array.push

File: packages/rocketchat-ui-admin/client/admin.js [hunk 6]
```diff
@@ -0,0 +1,668 @@
...
+	},
+	selectedOption(_id, val) {
+		const option = RocketChat.settings.collectionPrivate.findOne({ _id });
+		return option && option.value === val;
+	},
+	random() {
+		return Random.id();
+	},
+	getEditorOptions(readOnly) {
+		if (readOnly == null) {
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: ```getEditorOptions(readOnly = false)```

File: packages/rocketchat-ui-admin/client/admin.js [hunk 7]
```diff
@@ -0,0 +1,668 @@
...
+	assetAccept(fileConstraints) {
+		if (fileConstraints.extensions && fileConstraints.extensions.length) {
+			return `.${ fileConstraints.extensions.join(', .') }`;
+		}
+	},
+	autocompleteRoom() {
+		return {
+			limit: 10,
+			rules: [
+				{
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: could you put the comment back?

File: packages/rocketchat-ui-admin/client/admin.js [hunk 8]
```diff
@@ -0,0 +1,668 @@
...
+		};
+	},
+	selectedRooms() {
+		return Template.instance().selectedRooms.get()[this._id] || [];
+	},
+	getColorVariable(color) {
+		return color.replace(/theme-color-/, '@');
+	},
+	showResetButton() {
+		const setting = TempSettings.findOne({
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: one line

File: packages/rocketchat-ui-admin/client/admin.js [hunk 9]
```diff
@@ -0,0 +1,668 @@
...
+				packageValue: 1
+			}
+		});
+		return this.type !== 'asset' && setting.value !== setting.packageValue && !this.blocked;
+	}
+});
+
+Template.admin.events({
+	'change .input-monitor, keyup .input-monitor': _.throttle(function(e) {
+		let value;
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: you can declare and set the value at same line.

File: packages/rocketchat-ui-admin/client/admin.js [hunk 10]
```diff
@@ -0,0 +1,668 @@
...
+		const e = ev.originalEvent || ev;
+		let files = e.target.files;
+		if (!files || files.length === 0) {
+			if (e.dataTransfer && e.dataTransfer.files) {
+				files = e.dataTransfer.files;
+			} else {
+				files = [];
+			}
+		}
+		const results = [];
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: you dont need a result, coffee by default always returns the last thing.

File: packages/rocketchat-ui-admin/client/admin.js [hunk 11]
```diff
@@ -0,0 +1,668 @@
...
+		Object.keys(files).forEach((key) => {
+			const blob = files[key];
+			toastr.info(TAPi18n.__('Uploading_file'));
+			const reader = new FileReader();
+			reader.readAsBinaryString(blob);
+			results.push(reader.onloadend = () => {
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: it's wrong, see again.
```
reader.onloadend = =>
    Meteor.call 'setAsset', reader.result, blob.type, @asset, (err, data) ->
       if err?
           handleError(err)
 	# toastr.error err.reason, TAPi18n.__ err.error
 	  console.log err
 	  return
```

**Comment by MartinSchoeler** [comment]
@ggazzo or @rodrigok could you do a review?

File: packages/rocketchat-ui-admin/client/admin.js [hunk 12]
```diff
@@ -520,43 +515,38 @@ Template.admin.events({
...
 		event.currentTarget.value = '';
-		return event.currentTarget.focus();
+		event.currentTarget.focus();
 	},
 	'click .remove-room'(event, instance) {
 		const docId = this._id;
 		const settingId = event.currentTarget.getAttribute('data-setting');
 		const selectedRooms = instance.selectedRooms.get();
 		selectedRooms[settingId] = _.reject(selectedRooms[settingId] || [], function(setting) {
-			return setting._id === docId;
```
Comment (rodrigok) [inline packages/rocketchat-ui-admin/client/admin.js]: You can't remove this return

**Pull request merged**